### ISSUE TITLE
Not working in Gmail App for iOS

### ISSUE THREAD

**Issue by BaconDouglas** [ThemeMountain/pine#26]
In recent tests, we're finding that this no longer opens on the iOS Gmail app.

**Comment by cossssmin** [comment]
Can you please elaborate? Do you mean it isn't responsive... or... ?

**Comment by BaconDouglas** [comment]
It isn't.  When I click directly on it it does not respond/open.  Some apps require a click slightly to the right of the button and that isn't working.  Additionally, I've clicked all around it and it won't open.

**Comment by cossssmin** [comment]
Can you please share the code that you're using? Also, note that Gmail App used with a non-Gmail account does not support media queries/responsive design.

**Comment by BaconDouglas** [comment]
> Also, note that Gmail App used with a non-Gmail account does not support media queries/responsive design.

I did read up on this fact.  I've been testing into several gmail accounts ending in @gmail.com

The code I used in my most recent test was brought down from here this morning.  I'm including it below:

**The CSS**
```
@import url('https://fonts.googleapis.com/css?family=Merriweather|Open+Sans');
      @media only screen {
        .col, td, th, div, p {font-family: -apple-system,system-ui,BlinkMacSystemFont,"Segoe UI","Roboto","Helvetica Neue",Arial,sans-serif;}
        .serif {font-family: 'Merriweather', Georgia, serif!important;}
        .sans-serif {font-family: 'Open Sans', Arial, sans-serif!important;}
      }

      #outlook a {padding: 0;}
      img {border: 0; line-height: 100%; vertical-align: middle;}
      .col {font-size: 16px; line-height: 26px; vertical-align: top;}

      @media only screen and (max-width: 730px) {
        .wrapper img {max-width: 100%;}
        u ~ div .wrapper {min-width: 100vw;}
        .container {width: 100%!important; -webkit-text-size-adjust: 100%;}
      }

      @media only screen and (max-width: 699px) {
        .col {
          box-sizing: border-box;
          display: inline-block!important;
          line-height: 23px;
          width: 100%!important;
        }

        .col-sm-1  {max-width: 8.33333%;}
        .col-sm-2  {max-width: 16.66667%;}
        .col-sm-3  {max-width: 25%;}
        .col-sm-4  {max-width: 33.33333%;}
        .col-sm-5  {max-width: 41.66667%;}
        .col-sm-6  {max-width: 50%;}
        .col-sm-7  {max-width: 58.33333%;}
        .col-sm-8  {max-width: 66.66667%;}
        .col-sm-9  {max-width: 75%;}
        .col-sm-10 {max-width: 83.33333%;}
        .col-sm-11 {max-width: 91.66667%;}

        .col-sm-push-1  {margin-left: 8.33333%;}
        .col-sm-push-2  {margin-left: 16.66667%;}
        .col-sm-push-3  {margin-left: 25%;}
        .col-sm-push-4  {margin-left: 33.33333%;}
        .col-sm-push-5  {margin-left: 41.66667%;}
        .col-sm-push-6  {margin-left: 50%;}
        .col-sm-push-7  {margin-left: 58.33333%;}
        .col-sm-push-8  {margin-left: 66.66667%;}
        .col-sm-push-9  {margin-left: 75%;}
        .col-sm-push-10 {margin-left: 83.33333%;}
        .col-sm-push-11 {margin-left: 91.66667%;}

        .full-width-sm {display: table!important; width: 100%!important;}
        .stack-sm-first {display: table-header-group!important;}
        .stack-sm-last {display: table-footer-group!important;}
        .stack-sm-top {display: table-caption!important; max-width: 100%; padding-left: 0!important;}

        .toggle-content {
          max-height: 0;
          overflow: auto;
          transition: max-height .4s linear;
            -webkit-transition: max-height .4s linear;
        }
        .toggle-trigger:hover + .toggle-content,
        .toggle-content:hover {max-height: 999px!important;}

        .show-sm {
          display: inherit!important;
          font-size: inherit!important;
          line-height: inherit!important;
          max-height: none!important;
        }
        .hide-sm {display: none!important;}

        .align-sm-center {
          display: table!important;
          float: none;
          margin-left: auto!important;
          margin-right: auto!important;
        }
        .align-sm-left {float: left;}
        .align-sm-right {float: right;}

        .text-sm-center {text-align: center!important;}
        .text-sm-left   {text-align: left!important;}
        .text-sm-right  {text-align: right!important;}

        .borderless-sm {border: none!important;}
        .nav-sm-vertical .nav-item {display: block;}
        .nav-sm-vertical .nav-item a {display: inline-block; padding: 4px 0!important;}

        .spacer {height: 0;}

        .p-sm-0 {padding: 0!important;}
        .p-sm-10 {padding: 10px!important;}
        .p-sm-20 {padding: 20px!important;}
        .p-sm-30 {padding: 30px!important;}
        .pt-sm-0 {padding-top: 0!important;}
        .pt-sm-10 {padding-top: 10px!important;}
        .pt-sm-20 {padding-top: 20px!important;}
        .pt-sm-30 {padding-top: 30px!important;}
        .pr-sm-0 {padding-right: 0!important;}
        .pr-sm-10 {padding-right: 10px!important;}
        .pr-sm-20 {padding-right: 20px!important;}
        .pr-sm-30 {padding-right: 30px!important;}
        .pb-sm-0 {padding-bottom: 0!important;}
        .pb-sm-10 {padding-bottom: 10px!important;}
        .pb-sm-20 {padding-bottom: 20px!important;}
        .pb-sm-30 {padding-bottom: 30px!important;}
        .pl-sm-0 {padding-left: 0!important;}
        .pl-sm-10 {padding-left: 10px!important;}
        .pl-sm-20 {padding-left: 20px!important;}
        .pl-sm-30 {padding-left: 30px!important;}
        .px-sm-0 {padding-right: 0!important; padding-left: 0!important;}
        .px-sm-10 {padding-right: 10px!important; padding-left: 10px!important;}
        .px-sm-20 {padding-right: 20px!important; padding-left: 20px!important;}
        .px-sm-30 {padding-right: 30px!important; padding-left: 30px!important;}
        .py-sm-0 {padding-top: 0!important; padding-bottom: 0!important;}
        .py-sm-10 {padding-top: 10px!important; padding-bottom: 10px!important;}
        .py-sm-20 {padding-top: 20px!important; padding-bottom: 20px!important;}
        .py-sm-30 {padding-top: 30px!important; padding-bottom: 30px!important;}
      }
```

**The HTML**
```
<table bgcolor="#EEEEEE" cellpadding="15" cellspacing="0" class="wrapper" width="100%">
	<tbody>
		<tr>
			<td>
			<table align="center" bgcolor="#FFFFFF" cellpadding="0" cellspacing="0" class="container" width="700">
				<tbody>
					<tr>
						<td style="padding:30px 0">
						<table align="center" cellpadding="0" cellspacing="0" width="100%">
							<tbody>
								<tr>
									<td style="padding:0 20px">
									<table cellpadding="0" cellspacing="0" width="100%">
										<tbody>
											<tr><!-- Logo -->
												<td class="col pb-sm-2" style="padding:0 10px" width="200"><img alt="Logo" class="align-sm-center" src="http://placehold.it/160x160?text=LOGO" style="max-width:80px; width:100%" width="80" /></td>
												<!-- /Logo --><!-- Navigation -->
												<td class="col" style="padding:0 10px; vertical-align:middle" width="420">
												<div><!--[if !mso 9]><!--><img class="align-sm-center p-sm-1 show-sm toggle-trigger" src="https://i.imgur.com/RIRjWm6.png" style="display:none; max-width:20px; width:100%" width="20" /> <!--<![endif]-->
												<div class="toggle-content">
												<table align="right" cellpadding="0" cellspacing="0" class="align-sm-center nav nav-sm-vertical text-sm-center">
													<tbody>
														<tr>
															<td class="nav-item" style="padding-right:30px"><a href="http://example.com" style="color:#999999; text-decoration:none">News</a></td>
															<td class="nav-item" style="padding-right:30px"><a href="http://example.com" style="color:#999999; text-decoration:none">Services</a></td>
															<td class="nav-item" style="padding-right:30px"><a href="http://example.com" style="color:#999999; text-decoration:none">Work</a></td>
															<td class="nav-item"><a href="http://example.com" style="color:#999999; text-decoration:none">Store</a></td>
														</tr>
													</tbody>
												</table>
												</div>
												</div>
												</td>
												<!-- /Navigation -->
											</tr>
										</tbody>
									</table>
									</td>
								</tr>
							</tbody>
						</table>
						</td>
					</tr>
				</tbody>
			</table>
			</td>
		</tr>
	</tbody>
</table>
```

**Comment by cossssmin** [comment]
Sorry, I just understood what you mean 🤦‍♂️

So that menu button is using a `:hover` technique instead of a checkbox, mainly because I was building Pine when developing emails for MailChimp, and they don't allow any form elements.

Indeed, some email clients don't support `:hover`, rendering the button useless. Unfortunately I don't think there's any way around it with hover - you'd need to code your own hamburger menu for email, using checkboxes (unless you use MailChimp, in which case I have no clue).

For the ones that do support it, I see a typo in the mobile padding class on the menu image. This:

```html
<div><!--[if !mso 9]><!--><img class="align-sm-center p-sm-1 show-sm toggle-trigger" src="https://i.imgur.com/RIRjWm6.png" style="display:none; max-width:20px; width:100%" width="20" /> <!--<![endif]-->
```

should be:

```html
<div><!--[if !mso 9]><!--><img class="align-sm-center p-sm-10 show-sm toggle-trigger" src="https://i.imgur.com/RIRjWm6.png" style="display:none; max-width:20px; width:100%" width="20" /> <!--<![endif]-->
```

Basically, `p-sm-1` must be `p-sm-10`, so it gets some padding on mobile. You can also try increasing it, for example `p-sm-20` gives a bit more clickable space.

You can read more about email hamburger menus, here: 
http://freshinbox.com/blog/hybrid-collapsible-menu-for-email/

**Comment by BaconDouglas** [comment]
Will you be making an attempt to build a checkbox variation of this hamburger menu for use outside of MailChimp or does this issue close out with the suggestion for us to build it in-house?

It used to work in Gmail for iOS in our tests several months ago.  It was only recently when we were testing that we noticed it stopped working.  Maybe Gmail has changed something on their end with the app settings for iOS.  We still see it working in the Gmail app for android devices.

**Comment by cossssmin** [comment]
Yeah, it used to work when I implemented it, but `:hover` support has been since dropped in Gmail apps (only works in their webmail). 

I'm not planning on working on a checkbox hamburger menu, but if anyone is up for it, I would accept a PR that implements it cleanly with Pine classes.
